server/src/main/java/io/unitycatalog/server/persist/utils/FileUtils.java

```diff
@@ -181,20 +65,24 @@ private static URI adjustFileUri(URI fileUri) {
     return URI.create(uriString);
   }
 
-  public static String convertRelativePathToURI(String url) {
-    if (url == null) {
-      return null;
-    }
-    if (isSupportedCloudStorageUri(url)) {
-      return url;
-    } else {
-      return adjustFileUri(createURI(url)).toString();
+  public static String toStandardizedURIString(String inputPath) {
+    try {
+      // Check if the path is already a URI with a valid scheme
+      URI uri = new URI(inputPath);
+      // If it's a file URI, standardize it
+      if (uri.getScheme() != null) {
+        return switch (uri.getScheme()) {
+          case "file" -> adjustLocalFileURI(uri).toString();
+          case Constants.URI_SCHEME_S3, Constants.URI_SCHEME_ABFS, Constants.URI_SCHEME_ABFSS, Constants.URI_SCHEME_GS ->
```

Comment:
When would the scheme be S3? And why do we throw an exception for it?

Reply:
The scheme would be s3 when the UC OSS server is configured to work with a s3 bucket as a storage root. exception is thrown for any scheme other than file, s3, abfss, gs.